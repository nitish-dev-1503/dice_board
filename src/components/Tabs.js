import { useState } from "react";

import SevRoll from './SevRoll';
import Reach50 from './Reach50';
import TabItem from "./TabItem";
import GameSection from "./GameSection";


const Tabs = () => {
    const [activeTab, setActiveTab] = useState("tab1");

    return (
        <div>
            <ul className=" bg-primary_dark text-white font-medium flex w-1/2 m-auto items-center rounded-full">
                <TabItem title="SevRoll" id="tab1" activeTab={activeTab} setActiveTab={setActiveTab} />
                <TabItem title="Reach50" id="tab2" activeTab={activeTab} setActiveTab={setActiveTab} />
            </ul>

            <GameSection id="tab1" activeTab={activeTab}>
                <SevRoll />
            </GameSection>
            <GameSection id="tab2" activeTab={activeTab}>
                <Reach50 />
            </GameSection>

        </div>
    );
}

export default Tabs;