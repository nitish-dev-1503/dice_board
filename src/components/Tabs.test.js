import '@testing-library/jest-dom';
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Tabs from "./Tabs";


describe('Tabs components snapshot', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<Tabs />);
        
        expect(tree.toJSON()).toMatchSnapshot();
    })
})

describe('Tabs components', () => {
    test('renders two tab items', () => {
        render(<Tabs />);

        const tabItems = screen.getAllByRole("listitem");
        expect(tabItems.length).toBe(2);
    })

    test('renders SevRoll tab items', () => {
        render(<Tabs />);

        const tabItems = screen.getByText("SevRoll");
        expect(tabItems).toBeInTheDocument();
    })

    test('renders Reach50 tab items', () => {
        render(<Tabs />);

        const tabItems = screen.getByText("Reach50");
        expect(tabItems).toBeInTheDocument();
    })
})