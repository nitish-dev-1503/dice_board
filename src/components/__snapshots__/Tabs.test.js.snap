// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs components snapshot renders correctly 1`] = `
<div>
  <ul
    className=" bg-primary_dark text-white font-medium flex w-1/2 m-auto items-center rounded-full"
  >
    <li
      className="tab  bg-primary"
      onClick={[Function]}
    >
      SevRoll
    </li>
    <li
      className="tab false"
      onClick={[Function]}
    >
      Reach50
    </li>
  </ul>
  <div>
    <div
      className="game"
    >
      <img
        alt="sevroll-logo"
        className="h-20 my-10"
        src="sevroll.svg"
      />
      <div
        className="text-left bg-teal-50 p-6 rounded-lg m-4"
      >
        <p
          className="font-medium"
        >
          How to play?
        </p>
        <p>
          Roll the dice to score points and score a multiple of 7 to win.
        </p>
      </div>
      <p
        className="my-2 font-medium"
      >
        Your Score
      </p>
      <p
        className=" text-6xl text-secondary font-bold "
      >
        0
      </p>
      <button
        className="btn-primary"
        disabled={false}
        onClick={[Function]}
      >
        Roll the dice
      </button>
      <p>
        roll the dice to play
      </p>
      <div
        className="h-20"
      />
      <button
        className="btn-primary"
        disabled={true}
        onClick={[Function]}
      >
        Replay
      </button>
    </div>
  </div>
</div>
`;
